framework/src/application.ts

```diff
@@ -255,6 +264,11 @@ export class Application {
 	public async shutdown(errorCode = 0, message = ''): Promise<void> {
 		this.logger.info({ errorCode, message }, 'Application shutdown started');
 
+		// Wait if the loading process still in progress
+		if (this._loadingProcess) {
```

Comment:
it's better to use verbs/actions to make the code more readable though probably this is going to look better once it is replaced by the Jobs as suggested by Shu.